generator/cli/kallax/cmd/migrate.go

```diff
@@ -0,0 +1,195 @@
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/mattes/migrate"
+	_ "github.com/mattes/migrate/database/postgres"
+	_ "github.com/mattes/migrate/source/file"
+
+	"gopkg.in/src-d/go-kallax.v1/generator"
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+var Migrate = cli.Command{
+	Name:   "migrate",
+	Usage:  "Generate migrations for current kallax models",
+	Action: migrateAction,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "out, o",
+			Usage: "Output directory of migrations",
+		},
+		cli.StringFlag{
+			Name:  "name, n",
+			Usage: "Descriptive name for the migration",
+			Value: "migration",
+		},
+		cli.StringSliceFlag{
+			Name:  "input, i",
+			Usage: "List of directories to scan models from. You can use this flag as many times as you want.",
+		},
+	},
+	Subcommands: cli.Commands{
+		Up,
+		Down,
+	},
+}
+
+var migrationFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "dir, d",
+		Value: "./migrations",
+		Usage: "Directory where your migrations are stored",
+	},
+	cli.StringFlag{
+		Name:  "dsn",
+		Usage: "PostgreSQL data source name. Example: `user:pass@localhost:5432/database?sslmode=enable`",
+	},
+	cli.UintFlag{
+		Name:  "steps, n",
+		Usage: "Number of migrations to run",
+	},
+	cli.UintFlag{
+		Name:  "version, v",
+		Usage: "Migrate to a specific version. If `steps` and this flag are given, this will be used.",
+	},
+}
+
+var Up = cli.Command{
+	Name:   "up",
+	Usage:  "Executes the migrations from the current version until the specified version.",
+	Action: runMigrationAction(upAction),
+	Flags:  migrationFlags,
+}
+
+var Down = cli.Command{
+	Name:   "down",
+	Usage:  "Downgrades the database a certain number of migrations or until a certain version.",
+	Action: runMigrationAction(downAction),
+	Flags:  migrationFlags,
+}
+
+func upAction(m *migrate.Migrate, steps, version uint) error {
+	if version > 0 {
+		if err := m.Migrate(version); err != nil {
+			return fmt.Errorf("kallax: unable to upgrade up to version %d: %s", version, err)
+		}
+	} else if steps > 0 {
+		if err := m.Steps(int(steps)); err != nil {
+			return fmt.Errorf("kallax: unable to execute %d migration(s) up: %s", steps, err)
+		}
+	} else {
+		fmt.Println("WARN: No `version` or `steps` provided, upgrading all the way up.")
```

Comment:
Do this is a little bit dangerous, isn't it? I would prefer an error if no version or steps are provided, and if you want to update all, add an extra flag.